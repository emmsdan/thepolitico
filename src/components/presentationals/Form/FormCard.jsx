import React from 'react';
import PropTypes from 'prop-types';

const FormCard = props => {
  return (
    <form {...props} className="form-fields">
      <fieldset className="fieldset">
        <legend>{props.header}</legend>
        {props.children}
      </fieldset>
    </form>
  );
};

FormCard.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  header: PropTypes.any,
  children: PropTypes.any,
};

export default FormCard;
