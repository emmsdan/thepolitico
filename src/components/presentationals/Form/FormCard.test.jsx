import React from 'react';
import { shallow } from 'enzyme';

import FormCard from './FormCard';

describe('<FormCard />', () => {
  const header = 'The Sample Header';
  const Input = () => (
    <input className="my_sample_input" defaultValue="my awesome input sample" />
  );

  it('should render correctly', () => {
    const component = shallow(
      <FormCard header={header}>
        <Input />
      </FormCard>,
    );
    expect(component).toMatchSnapshot();
  });
});
