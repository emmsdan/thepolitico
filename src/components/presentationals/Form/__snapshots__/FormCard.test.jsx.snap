// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormCard /> should render correctly 1`] = `
<form
  className="form-fields"
  header="The Sample Header"
>
  <fieldset
    className="fieldset"
  >
    <legend>
      The Sample Header
    </legend>
    <Input />
  </fieldset>
</form>
`;
